### PR TITLE
[eks/argocd] Added ArgoCD notification configuration

### PR DESCRIPTION
**Pull request by goruha**
## what

* ArgoCD Github webhook integration
* ArgoCD Github Notification configuration
* Cleanup trunk configurations

## why

* Bugfix and improvements for ArgoCD sync mode

### CODE REVIEW

File: modules/argocd-repo/ssm.tf [hunk 1]
```diff
@@ -10,7 +10,7 @@ data "aws_ssm_parameter" "github_api_key" {
 
 module "store_write" {
   source  = "cloudposse/ssm-parameter-store/aws"
-  version = "0.11.0"
+  version = "0.8.3"
```
Comment (milldr) [inline modules/argocd-repo/ssm.tf:13]: revert change

File: modules/argocd-repo/ssm.tf [hunk 2]
```diff
@@ -22,5 +22,5 @@ module "store_write" {
     }
   ]
 
-  context = module.this.context
+  context = module.introspection.context
```
Comment (milldr) [inline modules/argocd-repo/ssm.tf:25]: revert change

File: modules/eks/argocd/data.tf [hunk 3]
```diff
@@ -33,16 +25,6 @@ data "aws_ssm_parameter" "oidc_client_secret" {
...
-#data "aws_eks_cluster" "kubernetes" {
-#  count = local.count_enabled
-#  name  = module.eks.outputs.eks_cluster_id
-#}
-
-#data "aws_eks_cluster_auth" "kubernetes" {
-#  count = local.count_enabled
-#  name  = module.eks.outputs.eks_cluster_id
-#}
-
```
Comment (milldr) [inline modules/eks/argocd/data.tf:45]: remove commented code if not applicable

File: modules/eks/argocd/provider-github.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+data "aws_ssm_parameter" "github_api_key" {
+  count           = local.github_webhook_enabled ? 1 : 0
+  name            = var.ssm_github_api_key
+  with_decryption = true
+
+  provider = aws.config_secrets
+}
+
+provider "github" {
+  base_url = local.github_webhook_enabled ? var.github_base_url : null
```
[the remark this answers is not in the document]
Comment (goruha) [reply modules/eks/argocd/provider-github.tf:32]: `enabled` used in `local.github_webhook_enabled`

File: modules/argocd-repo/ssm.tf (hunk 2 above)
Comment (goruha) [reply modules/argocd-repo/ssm.tf:25]: I thought we deprecated `introspection` mixing using.
Am I wrong?

Comment (milldr) [reply modules/argocd-repo/ssm.tf:25]: we did deprecate introspection, but this change is adding it back

File: modules/argocd-repo/providers.tf [hunk 5]
```diff
@@ -8,7 +8,7 @@ provider "aws" {
     # module.iam_roles.terraform_role_arn may be null, in which case do not assume a role.
     for_each = compact([module.iam_roles.terraform_role_arn])
     content {
-      role_arn = assume_role.value
+      role_arn = module.iam_roles.terraform_role_arn
```
[the remark this answers is not in the document]
Comment (goruha) [reply modules/argocd-repo/providers.tf:11]: This is the new pattern. 
Check all latest components `provider.tf`

[the remark this answers is not in the document]
Comment (goruha) [reply modules/argocd-repo/providers.tf:11]: Done

**Pull request merged**